tests/functional/relation_tests/test_relation_type_change.py

```diff
@@ -1,21 +1,26 @@
 from dataclasses import dataclass
 from itertools import product
+from typing import Optional
 
 from dbt.tests.util import run_dbt
 import pytest
 
 from tests.functional.relation_tests import models
-from tests.functional.utils import query_relation_type, update_model
+from tests.functional.utils import describe_dynamic_table, query_relation_type, update_model
 
 
 @dataclass
 class Model:
     model: str
     relation_type: str
+    table_format: Optional[str] = None
 
     @property
     def name(self) -> str:
-        return f"{self.relation_type}"
+        name = f"{self.relation_type}"
+        if self.table_format:
+            name += f"_{self.table_format}"
+        return name
```

Comment:
It's a trifle in some respect, but I'd aim for a more pythonic solution here.
```suggestion
        if self.table_format:
            return f"{self.relation_type}_{self.table_format}"
        else:
            return f"{self.relation_type}"
```

or even 
```suggestion
    def get_name(self):
        parts = [self.relation_type]
        if self.table_format:
            parts.append(self.table_format)
        return '_'.join(parts)
```

Mutating strings feels unidiomatic so I tend to call it out when I see as a chance to avoid code smell.

Reply:
I don't think of building strings as "un-pythonic", but if it makes it easier to read what's going on (which I see the argument here), I don't mind changing it. However, I wouldn't have separate `return`s. I would create the name in each branch and then return the name at the end. I'll make that change.

Reply:
Fair enough. Maybe I've just been reading too much new fangled functional Python books of late which I totally accept is an arbitrary choice :)